### PR TITLE
docs(blog-post): replicate spatial dev guru blog

### PR DESCRIPTION
**Pull request by ncclementi**
## Description of changes
Replicate Spatial Dev Guru's blog using Ibis

## Issues closed

Closes https://github.com/ibis-project/ibis/issues/7717

### CODE REVIEW

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (cpcloud) [inline docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd:60]:
```suggestion
# read data and rename columns to use snake case
```

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (cpcloud) [inline docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd:65]:
```suggestion
```

**Comment by ncclementi** [comment]
🤔 : I see on the preview this extra black bar, not sure why. 

Screenshot 2024-01-12 at 12 55 57 PM

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (cpcloud) [inline docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd:73]:
```suggestion
points as `YX` (latitude/longitude) while in Ibis, we assume data in the form
```

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (cpcloud) [inline docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd:76]: Missing some text?

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (cpcloud) [inline docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd:87]: These line are pretty long. Can you use `_` to condense them a bit? Also should be formatted.

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (cpcloud) [inline docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd:100]:
```suggestion
biketrip.start_station_name.topk(10, by=_.trips.count())
```

Does that produce the same result?

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (cpcloud) [inline docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd:106]:
```suggestion
biketrip.end_station_name.topk(10, by=_.trips.count())
```

Might work, similar to the `start_station_name` variant.

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (cpcloud) [inline docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd:120]:
```suggestion
    avg_distance=biketrip.pickup_point.distance(biketrip.dropoff_point).mean()
```

No reason to call `.name` in between other expressions.

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (cpcloud) [inline docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd:131]:
```suggestion
trip_distance = biketrip.pickup_point.distance(biketrip.dropoff_point)
```

No need to name this.

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (cpcloud) [inline docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd:133]: Can you format this line?

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (cpcloud) [inline docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd:159]: Can you format this line?

**Review by cpcloud — CHANGES_REQUESTED** [review]
Looking good! Mostly stylistic comments!

**Comment by cpcloud** [comment]
> 🤔 : I see on the preview this extra black bar, not sure why.

I think this is an artifact of DuckDB's progress bar.

You can turn it off with 

```sql
SET enable_progress_bar = true;
```

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (ncclementi) [reply docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd:73]: whoops, I'll fix that.

**Comment by deepyaman** [comment]
Minor comment--before it get's merged, can you make sure it's "spatial" instead of "sapatial". :)

File: docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd
Comment (ncclementi) [reply docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd]: nice, I wasn't aware of the `topk()` function, the equivalent query would be
`biketrip.end_station_name.topk(10)` In the first case I left both options to showcase the similarity.

**Comment by ncclementi** [comment]
@deepyaman great catch!

**Comment by ncclementi** [comment]
@cpcloud I think I cover all your comments/requests.  About the progress bar, I wasn't able to figure out where to set the variable `SET enable_progress_bar = false`?

**Comment by cpcloud** [comment]
Ah, sorry, that's code to execute with `con.raw_sql`

**Review by cpcloud — APPROVED** [review]

**Comment by cpcloud** [comment]
@ncclementi I can merge this now if you want to follow up with a PR to disable the progress bar, or happy to wait as well.

**Comment by cpcloud** [comment]
SGTM!

**Comment by gforsyth** [comment]
SHIP IT

**Pull request merged**
